imod/protocols/protocol_xcorr.py

```diff
@@ -0,0 +1,245 @@
+                        "-goutput %(goutput)s"
+            self.runJob('xftoxg', argsXftoxg % paramsXftoxg, cwd=workingFolder)
+
+        # Generate output tilt series
+        outputSetOfTiltSeries = self.getOutputSetOfTiltSeries()
+
+        for ts in self.inputSetOfTiltSeries.get():
+            tsId = ts.getTsId()
+            alignmentMatrix = self.formatTransformationMatrix(self._getExtraPath('%s/%s.prexg' % (tsId, tsId)))
+            newTs = tomoObj.TiltSeries(tsId=tsId)
+            newTs.copyInfo(ts)
+            outputSetOfTiltSeries.append(newTs)
+            i = 0
+
+            # For each tilt image in the series, assign its transform matrix
+            for tiltImage in ts:
+                newTi = tomoObj.TiltImage()
+                newTi.copyInfo(tiltImage, copyId=True)
+                newTi.setLocation(tiltImage.getLocation())
+
+                # Set the tansformation matrix
+                transform = data.Transform()
+                transform.setMatrix(alignmentMatrix[:, :, i])
+                newTi.setTransform(transform)
+                i += 1
+                newTs.append(newTi)
+            outputSetOfTiltSeries.update(newTs)  # update items and size info
+        self._store()
+
+    def computeInterpolatedStackStep(self):
+        outputInterpolatedSetOfTiltSeries = self.getOutputInterpolatedSetOfTiltSeries()
+
+        for ts in self.inputSetOfTiltSeries.get():
+            tsId = ts.getTsId()
+            newTs = tomoObj.TiltSeries(tsId=tsId)
+            newTs.copyInfo(ts)
+            outputInterpolatedSetOfTiltSeries.append(newTs)
+            workingFolder = self._getExtraPath(tsId)
+            paramsAlginment = {
+                'input': "%s.st" % tsId,
+                'output': '%s_preali.st' % tsId,
+                'xform': "%s.prexg" % tsId,
+                'bin': int(self.binning.get()),
+                'mode': 0,
+                'float': 2,
+                'imagebinned': 1.0}
+            argsAlignment = "-input %(input)s " \
+                            "-output %(output)s " \
+                            "-xform %(xform)s " \
+                            "-bin %(bin)d " \
+                            "-mode %(mode)s " \
+                            "-float %(float)s " \
+                            "-imagebinned %(imagebinned)s"
+
+            self.runJob('newstack', argsAlignment % paramsAlginment, cwd=workingFolder)
+            for tiltImage in ts:
+                newTi = tomoObj.TiltImage()
+                newTi.copyInfo(tiltImage, copyId=True)
+                newTi.setLocation(os.path.join(workingFolder, '%s_preali.st' % tsId))
+                newTs.append(newTi)
+            if self.binning > 1:
+                newTs.setSamplingRate(ts.getSamplingRate()*int(self.binning.get()))
+            outputInterpolatedSetOfTiltSeries.update(newTs)  # update items and size info
+        self._store()
+
+    def _createOutputStep(self):
+        pass
```

Comment:
Why there is a createOutputStep empty. Either do not have or have it with content. I believe the reason for having a final createOutputStep is to avoid several steps (in case of parallelization) to access the same set db avoiding concurrency problems.

If this protocol is not meant to run in parallel, I guess we can resume this way but must keep an eye on this approach.

Reply:
I'd remove this method and it's insertion

Reply:
BTW, Good job @fede-pe